scripts/generate_leaderboard.py

```
@@ -161,15 +192,27 @@ def is_within_date_range(
     return True
 
 
-def build_leaderboard(start_date: datetime, end_date: datetime):
+def build_leaderboard(start_date: datetime, end_date: Optional[datetime]):
```

Ruff (0.15.5)

[warning] Too many branches (15 > 12)

(PLR0912)